ecs-agent/utils/execwrapper/exec.go

```diff
@@ -0,0 +1,91 @@
+package execwrapper
+
+import (
+	"context"
+	"io"
+	"os"
+	"os/exec"
+)
+
+// Exec acts as a wrapper to functions exposed by the exec package.
+// Having this interface enables us to create mock objects we can use
+// for testing.
+type Exec interface {
+	CommandContext(ctx context.Context, name string, arg ...string) Cmd
+}
+
+// execWrapper is a placeholder struct which implements the Exec interface.
+type execWrapper struct {
+}
+
+func NewExec() Exec {
+	return &execWrapper{}
+}
+
+// CommandContext essentially acts as a wrapper function for exec.CommandContext function.
+func (e *execWrapper) CommandContext(ctx context.Context, name string, arg ...string) Cmd {
+	return NewCMDContext(ctx, name, arg...)
+}
+
+// Cmd acts as a wrapper to functions exposed by the exec.Cmd object.
+// Having this interface enables us to create mock objects we can use
+// for testing.
+type Cmd interface {
+	Run() error
+	Start() error
+	Wait() error
+	KillProcess() error
+	AppendExtraFiles(...*os.File)
+	Args() []string
+	SetIOStreams(io.Reader, io.Writer, io.Writer)
+}
+
+type cmdWrapper struct {
+	*exec.Cmd
+}
```

Comment:
Why do we need to create this new `struct`? Shouldn't `exec.Cmd` already implement `Cmd` interface?

Reply:
The `cmdWrapper` interface is not exactly the same, we also added a couple helper functions to help with common used functionalities such as `SetIOStreams()`. Without having this extra struct, we won't be able to make the helper functions non-static (we would get error `cannot define new methods on non-local type exec.Cmd`)